packages/api/src/graphql/query.public.ts

```diff
@@ -259,7 +259,7 @@ export const GraphQLPublicQuery = new GraphQLObjectType<undefined, Context>({
       async resolve(root, {id, slug, token}, {session, loaders, verifyJWT}) {
         let page = id ? await loaders.publicPagesByID.load(id) : null
 
-        if (!page) {
+        if (!page && slug !== undefined) {
```

Comment:
Yes, it works. It's because you want to be sure there's a slug but it can still be an empty string.
![slug_query](https://user-images.githubusercontent.com/31937943/179086394-fd31649f-eafe-4d71-84ec-5a10ce84140a.jpeg)

Reply:
But shouldn't that have worked before?

Reply:
Yes with slug it worked, but it entered this condition even when there was no slug and threw an error expecting one. With this check, it will skip the query by slug when there's no slug and go to the query by token. Its not `!slug` to allow for empty string slug.

Reply:
![tokenquery](https://user-images.githubusercontent.com/31937943/179460011-4a8a196f-ef89-4d40-bb82-46d900fed4c0.png)